src/Core/Notifications/DoNotShowAgainNotificationAction.cs

```diff
@@ -0,0 +1,51 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+
+namespace SonarLint.VisualStudio.Core.Notifications
+{
+    public interface IDoNotShowAgainNotificationAction : INotificationAction
+    {
+    }
+
+    public class DoNotShowAgainNotificationAction : IDoNotShowAgainNotificationAction
+    {
+        private readonly IDisabledNotificationsStorage disabledNotificationsStorage;
+
+        public DoNotShowAgainNotificationAction(IDisabledNotificationsStorage disabledNotificationsStorage)
+        {
+            this.disabledNotificationsStorage = disabledNotificationsStorage;
+        }
+
+        public string CommandText => "Do not show again";
```

Comment:
The wording in the VS gold bars is more informal e.g. `Don't report this extension again`.
Something like `Don't show this again` would be closer to that VS style.

It's a UI string, so it should be in a resource file.